### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by AdamWill**
Several changes to make the tests work properly.

### CODE REVIEW

**Comment by AdamWill** [comment]
Yeah, it looks like Actions aren't turned on for the project yet. If someone could do that it'd be great. I want to merge this so we can get CI for the other pending PRs, it will help with reviewing them.

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+name: Python package with Tox
```
Comment (waldyrious) [inline .github/workflows/tox.yml:2]: Let's make this a bit more explicit :)

```suggestion
name: Test the Python package with Tox
```

File: setup.cfg [hunk 2]
```diff
@@ -23,4 +23,4 @@ addopts = --cov mwclient test
 [flake8]
 max-line-length = 90
 ignore = 
-	W503  # Line break before binary operator
```
Comment (waldyrious) [inline setup.cfg:26]: According to https://flake8.pycqa.org/en/latest/user/configuration.html, inline comments are not supported, but they can be added in a separate line. Can you do that, please? I'd prefer if we preserved the comment.

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,py35,py36,py37,py38,flake
+envlist = py27,py35,py36,py37,py38,py39,py310,py311,py312,flake
```
Comment (waldyrious) [inline tox.ini:2]: Would it be worth dropping the [EOL'd Python versions](https://devguide.python.org/versions/)? For the record, I don't have anything against ensuring the project keeps running in old versions, as long as that doesn't compromise our ability to adopt new features or cleanup the codebase.

**Comment by waldyrious** [comment]
I realize the motivation for the `test/__init__.py` file is explained in the commit message (9f0e0adb193786df276087a7139531f5384f5a49), but perhaps it would be convenient to also add a comment to the same effect within the file?

**Comment by AdamWill** [comment]
In general (for your last two notes) I am not caring too hard about cleaning up things like that in this PR because I'm planning to propose much more change later - moving configuration into `pyproject.toml` as far as possible, using diff-cover and diff-quality and black and dropping flake8, using isolated builds in tox...this PR was meant to be more "minimum change necessary to get CI working with the existing stuff so we can review existing PRs and have a base for larger changes".

File: tox.ini (hunk 3 above)
Comment (AdamWill) [reply tox.ini:2]: See comment in main thread - I agree, but kinda planned to do that later. I'm going to suggest dropping Python 2 support entirely, and along with that I'd usually do some stuff like removing any now-unnecessary compat stuff for Python 2 (e.g. use of six), switching to format strings...

**Comment by AdamWill** [comment]
So I don't remember this from the last time I setup Actions, and I can't find a really clear reference, but I *think* when a PR adds a workflow, that workflow won't be run on that PR (which is a bit annoying, other CI systems I've used *do* do this). I think I'll split that commit out into another PR we can merge on its own, and then hopefully we can see how the workflow looks when it runs against this PR.

**Comment by AdamWill** [comment]
https://github.com/mwclient/mwclient/pull/282 has the Actions change.

**Comment by AdamWill** [comment]
OK, looks like the CI is working, whee. (I didn't include Python 2 in the CI stuff since I'm gonna propose dropping it anyway, but I did run it locally and it passes there too). If we can merge this, we can then rebase all other existing PRs and get CI run on them.

File: tox.ini (hunk 3 above)
Comment (AdamWill) [reply tox.ini]: gonna mark this as resolved and we'll address it later.

File: setup.cfg (hunk 2 above)
Comment (AdamWill) [reply setup.cfg:26]: done.

**Review by waldyrious — APPROVED** [review]
LGTM, thanks for taking this on! :tada:

**Comment by waldyrious** [comment]
@AdamWill I merged the PR with the "merge commit" strategy. I think that doing so, while ensuring branches are fully rebased on the tip of `master` prior to the merge, leads to the most readable git history graph, where multiple commits from the same branch are kept atomically separated but still bundled into the same aggregate change unit. In other words, this is the semi-linear git branching strategy. See https://github.com/isaacs/github/issues/1017 for more details and a diagram.

Let me know if you have any thoughts on this matter :)

**Comment by AdamWill** [comment]
Personally I prefer just rebase merging everything for my own projects and that's the most common strategy I run into in other projects, but there's a lot of tastes out there and I don't really mind a whole lot :)

**Comment by AdamWill** [comment]
I tried to merge https://github.com/mwclient/mwclient/pull/267 the way you like, did I get it right?

**Comment by waldyrious** [comment]
Yep, that's perfect, thanks!

**Pull request merged**